### PR TITLE
Order routes #21

### PR DESCRIPTION
**Pull request by Nitiquita**

### CODE REVIEW

File: db/models/order.js [hunk 1]
```diff
@@ -15,4 +17,5 @@ module.exports = db => db.define('orders', {
 
 module.exports.associations = (Order, {User, Product}) => {
   Order.belongsTo(User)
+  // Order.hasMany(Product, {through: 'OrderDetail'})
```
Comment (stella-yc) [inline db/models/order.js]: remove comment

File: server/auth.filters.js [hunk 2]
```diff
@@ -15,14 +15,28 @@ const selfOnly = action => (req, res, next) => {
...
+// }
+
 const assertAdmin = (req, res, next) => {
   if (!req.user) {
-    return res.status(401).send('You must be logged in');
-  } else if (!req.user.isAdmin) {
-    return res.status(403).send('You are unauthorized to view this page')
-  } 
-  next();
+    var err = new Error('not logged in')
```
Comment (stella-yc) [inline server/auth.filters.js]: remove, Allie is working on this

File: server/orders.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+const CartDetail = db.model('cart_details')
+const Product = db.model('products')
+
+const {mustBeLoggedIn, assertAdmin, selfOnly} = require('./auth.filters')
+
+module.exports = require('express').Router()
+
+// get all orders (Logged in User, ADMIN)
+  .get('/', mustBeLoggedIn, (req, res, next) => {
+    if (req.user.isAdmin === true) {
```
Comment (stella-yc) [inline server/orders.js]: just make it 
if (req.user.isAdmin)
for consistency

File: server/orders.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      Order.findAll({})
+        .then(orders => {
+          res.json(orders)
+        })
```
Comment (stella-yc) [inline server/orders.js]: Add a catch here!

File: server/orders.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      include: [{
+        model: User,
+        where: {
+          id: req.user.id
+        }
+      }]
+    }).then(orders => {
+      res.json(orders)
+    }).catch(next)
+    Order.findAll({
```
Comment (stella-yc) [inline server/orders.js]: As it is, we would never reach this Order.findAll for req.query.
PERHAPS Use an if statement earlier checking if there is req.query, maybe.

File: server/auth.filters.js [hunk 6]
```diff
@@ -69,11 +69,11 @@ const assertAdmin = (req, res, next) => {
   if (!req.user) {
     const err = new Error('You must be logged in')
     err.status = 401
-    throw err
+    next(err)
```
Comment (stella-yc) [inline server/auth.filters.js:72]: Is this a style thing? What is advantage of next vs throw?

File: server/orders.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    Cart.findOne({
+      where: {
+        user_id: req.user.id
+      },
+      include: [{
+        model: Product
+      }]
+    }).then(order => {
+      return res.json(order)
+    }).then(cart => Order.create({
```
Comment (stella-yc) [inline server/orders.js:55]: Does this get to the Order.create line?
Also, you find the Cart, but how do we update the Order Detail model to include what used to be in Cart Detail?

File: server/orders.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+      status: 'received',
+      user_id: req.user.id
+    })).catch(next)
+  })
+
+  .param('oid', (req, res, next, oid) => {
```
Comment (stella-yc) [inline server/orders.js:61]: When you look for an Order, consider including Product? That way we can see all the products within that order

**Review by stella-yc — CHANGES_REQUESTED** [review]
Thank you Nitiquita! I included some comments, give me a call if you want to talk it out!

File: server/auth.filters.js (hunk 6 above)
Comment (Nitiquita) [reply server/auth.filters.js:72]: I think next is better to use in express from what I'm looking at online, but we still need an error handler to deal with it.

File: server/orders.js (hunk 8 above)
Comment (Nitiquita) [reply server/orders.js:61]: Good call! I'll add it.

Comment (Nitiquita) [reply server/orders.js:61]: Unfortunately, they are not associated...

**Comment by Nitiquita** [comment]
Notes: the difference between OrderDetail and order-details tables?

**Comment by stella-yc** [comment]
Sorry I just saw this email!

Try deleting all the tables in your database, and then doing npm run dev
and then npm run seed. I have to keep doing this every time we change
models, since I don't think postico will delete old ones?

-Stella Chung

File: server/auth.filters.js (hunk 6 above)
Comment (stella-yc) [reply server/auth.filters.js:72]: cool!

**Review by stella-yc — APPROVED** [review]

**Pull request merged**